src/services/jsDoc.ts

```diff
@@ -188,24 +190,26 @@ namespace ts.JsDoc {
      * be performed.
      */
     export function getDocCommentTemplateAtPosition(newLine: string, sourceFile: SourceFile, position: number): TextInsertion {
+        const emptyDocComment = { newText: "", caretOffset: 0 };
+
         // Check if in a context where we don't want to perform any insertion
         if (isInString(sourceFile, position) || isInComment(sourceFile, position) || hasDocComment(sourceFile, position)) {
-            return undefined;
+            return emptyDocComment;

```

Comment:
so why do we return an empty object here? what is wrong in undefined?

Reply:
Was @andy-ms's suggestion. Andy, could you elaborate?

Reply:
See https://github.com/Microsoft/TypeScript/pull/19544#discussion_r147851114

Reply:
aha, I misunderstood what you meant by empty doc comment template.

Reply:
so how is the empty file case related to `isInString` or `isIncomment` ?

Reply:
if you want to enable them all the time that is fine, but not in strings and not in comments